### PR TITLE
feat(platform/gitea): support native automerge

### LINKED ISSUE #15968: Support Gitea for platform automerge
### What would you like Renovate to be able to do?

Gitea v1.17 will have a new `merge_when_checks_succeed` feature like github and gitlab. So i like to support this.
It can alsread ybe tried on `try.gitea.io`

### If you have any ideas on how this should be implemented, please tell us here.

call `merge` with `merge_when_checks_succeed=true`

https://try.gitea.io/api/swagger#/repository/repoMergePullRequest

### Is this a feature you are interested in implementing yourself?

Maybe

### LINKED ISSUE #16883: Support platformautoMerge on Gitea v1.17.0+
### What would you like Renovate to be able to do?

Gitea `v1.17.0` has now an automerge feature, so we should support that via `platformautoMerge`

- https://github.com/go-gitea/gitea/pull/9307

### If you have any ideas on how this should be implemented, please tell us here.

Doing the same as in github (trying to automerge after PR creation).

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by viceice**
## Changes
 Support native automerge on Gitea since v1.17.0

## Context
- closes #15968
- closes #16883
- https://github.com/go-gitea/gitea/pull/9307

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitea/index.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Either the account should have full name and email address set to allow Renovate
 ## Unsupported platform features/concepts
 
 - **Adding reviewers to PRs not supported**: Gitea versions older than v1.14.0 do not have the required API.
+- **Setting platformAutomerge To PRs not supported**: Gitea versions older than v1.17.0 do not have the required API.
```
Comment (HonkingGoose) [inline lib/modules/platform/gitea/index.md:22]:
```suggestion
- **Adding reviewers to PRs not supported**: Gitea versions older than v1.14.0 do not have the required API.
- **Setting `platformAutomerge` to use platform-native automerge for PRs not supported**: Gitea versions older than v1.17.0 do not have the required API.
```

How about this? I think we should use `monospace` styling for the config option name, and explain a bit better what `platformAutomerge` means.

File: lib/modules/platform/gitea/index.spec.ts [hunk 2]
```diff
@@ -1020,6 +1020,98 @@ describe('modules/platform/gitea/index', () => {
...
+        mockRepo.full_name,
+        mockNewPR.number,
+        {
+          Do: 'rebase',
+          merge_when_checks_succeed: true,
+        }
+      );
+    });
+
+    it("doesn't support platform automerge", async () => {
```
Comment (HonkingGoose) [inline lib/modules/platform/gitea/index.spec.ts:1091]: For me, reading the description, it's not clear what this is testing. I think the final `expect` is the wanted result?

```ts
     expect(helper.mergePR).not.toHaveBeenCalled();
```

Maybe you should rewrite the description to something like this:

```ts
   it("if Gitea does not support platform automerge then ....", async () => {
```

File: lib/modules/platform/gitea/index.ts [hunk 3]
```diff
@@ -495,6 +496,33 @@ const platform: Platform = {
...
+          } catch (err) {
+            logger.warn(
+              { err, prNumber: gpr.number },
+              'Gitea-native automerge: fail'
+            );
+          }
+        } else {
+          logger.debug(
+            { prNumber: gpr.number },
+            'Gitea-native automerge: not supported on this Gitea version. Requires >=1.17.0'
```
Comment (HonkingGoose) [inline lib/modules/platform/gitea/index.ts:521]:
```suggestion
            'Gitea-native automerge: not supported on this version of Gitea. Use 1.17.0 or newer.'
```

How about this? It think it explains better what the user can do to get things working: 

> Use version 1.17.0 or newer.

**Review by HonkingGoose — COMMENTED** [review]
Here's your review. 😉

File: lib/modules/platform/gitea/index.spec.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/platform/gitea/index.spec.ts:1091]:
```suggestion
    it('continues if platform automerge is not supported', async () => {
```
How about this?

Comment (HonkingGoose) [reply lib/modules/platform/gitea/index.spec.ts:1091]: I like it! 😄

File: lib/modules/platform/gitea/index.ts (hunk 3 above)
Comment (HonkingGoose) [reply lib/modules/platform/gitea/index.ts]: I've opened a PR to change the debug message for GitHub:

- #16889

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -1912,6 +1912,8 @@ Normally when you set `rebaseWhen=auto` Renovate rebases any branch that's behin
 This behavior is no longer guaranteed when you enable `platformAutomerge` because the platform might automerge a branch which is not up-to-date.
 For example, GitHub might automerge a Renovate branch even if it's behind the base branch at the time.
 
+Please check platform specific docs for version requirements.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1915]: Some of our platform docs do not mention what versions users should use for "full support for all Renovate features":

| Platform docs for | Mentions fully supported version? | Has section "supported versions" |
| --- | --- | --- |
| Azure | no | no |
| Bitbucket Cloud | no | no |
| Bitbucket Server | We don't support EOL versions | yes |
| Gitea | no, only mentions version for a single feature | no |
| GitHub and GitHub Enterprise Server | mentions GHES token version | no |
| GitLab | Mentions server version dependent features | no |

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:1915]: I have opened an issue to discuss this problem further:

- #16937

**Pull request merged**